File: src/lib/vagas/gerador.ts
/** Gera a descrição profissional da vaga a partir das respostas do chatbot. */

import type { RespostasVaga } from "./perguntas";

function listar(texto: string): string[] {
  return texto
    .split(/[;\n]|,(?![^()]*\))/g)
    .map((item) => item.trim().replace(/^[-•]\s*/, ""))
    .filter((item) => item.length > 0);
}

function capitalizar(texto: string): string {
  if (texto.length === 0) return texto;
  return texto.charAt(0).toUpperCase() + texto.slice(1);
}

export function gerarDescricao(respostas: RespostasVaga): string {
  const {
    cargo,
    area,
    nivel,
    tecnologias,
    responsabilidades,
    requisitos,
    beneficios,
    modeloTrabalho,
  } = respostas;

  const resumo = `Estamos buscando um(a) ${cargo} ${nivel ? `de nível ${nivel.toLowerCase()} ` : ""}para atuar na área de ${area.toLowerCase()}. A pessoa selecionada participará do desenvolvimento e da evolução de soluções, trabalhando com ${tecnologias.toLowerCase()} em um ambiente colaborativo e orientado a resultados.`;

  const blocos = [
    `# ${cargo}`,
    "",
    "## Resumo da oportunidade",
    resumo,
    "",
    "## Responsabilidades",
    ...listar(responsabilidades).map((item) => `- ${capitalizar(item)}`),
    "",
    "## Requisitos",
    ...listar(requisitos).map((item) => `- ${capitalizar(item)}`),
    "",
    "## Tecnologias",
    ...listar(tecnologias).map((item) => `- ${capitalizar(item)}`),
    "",
    "## Benefícios",
    ...listar(beneficios).map((item) => `- ${capitalizar(item)}`),
    "",
    "## Modelo de trabalho",
    capitalizar(modeloTrabalho),
  ];

  return blocos.join("\n").trim();
}

File: python/gerador.py
"""Gerador automático da descrição de vaga a partir das respostas do chatbot."""

from __future__ import annotations

import re

CAMPOS = (
    ("cargo", "Cargo da vaga"),
    ("area", "Área de atuação"),
    ("nivel", "Nível de experiência"),
    ("tecnologias", "Tecnologias necessárias"),
    ("responsabilidades", "Responsabilidades"),
    ("requisitos", "Requisitos"),
    ("beneficios", "Benefícios"),
    ("modelo_trabalho", "Modelo de trabalho"),
)


def _itens(texto: str) -> list[str]:
    partes = re.split(r"[;,\n]", texto)
    return [p.strip().lstrip("-• ").capitalize() for p in partes if p.strip()]


def gerar_descricao(respostas: dict[str, str]) -> str:
    """Monta a descrição profissional organizada por seções."""
    cargo = respostas.get("cargo", "")
    area = respostas.get("area", "")
    nivel = respostas.get("nivel", "")
    tecnologias = respostas.get("tecnologias", "")

    resumo = (
        f"Estamos buscando um(a) {cargo}"
        + (f" de nível {nivel.lower()}" if nivel else "")
        + f" para atuar na área de {area.lower()}. A pessoa selecionada participará do "
        f"desenvolvimento e da evolução de soluções, trabalhando com {tecnologias.lower()} "
        "em um ambiente colaborativo e orientado a resultados."
    )

    linhas = [f"# {cargo}", "", "## Resumo da oportunidade", resumo]

    for chave, titulo in (
        ("responsabilidades", "Responsabilidades"),
        ("requisitos", "Requisitos"),
        ("tecnologias", "Tecnologias"),
        ("beneficios", "Benefícios"),
    ):
        linhas += ["", f"## {titulo}"]
        linhas += [f"- {item}" for item in _itens(respostas.get(chave, ""))]

    linhas += ["", "## Modelo de trabalho", respostas.get("modelo_trabalho", "").capitalize()]

    return "\n".join(linhas).strip()
